index.js

```diff
@@ -237,7 +237,23 @@ const fileType = input => {
 		// - one entry named '[Content_Types].xml' or '_rels/.rels',
 		// - one entry indicating specific type of file.
 		// MS Office, OpenOffice and LibreOffice may put the parts in different order, so the check should not rely on it.
-		const findNextZipHeaderIndex = (buffer, startAt = 0) => buffer.indexOf(zipHeader, startAt);
+		const findNextZipHeaderIndex = (buffer, startAt = 0) => {
+			if (Buffer && Buffer.isBuffer(buffer)) {
+				return buffer.indexOf(Buffer.from(zipHeader), startAt);
+			}
+
+			// Uint8Array.indexOf, unlike Buffer.indexOf, can search for single byte only
+			let index = buffer.indexOf(zipHeader[0], startAt);
+			while (index >= 0) {
+				if (check(zipHeader, {offset: index})) {
+					return index;
+				}
+
+				index = buffer.indexOf(zipHeader[0], index + 1);
+			}
+
+			return -1;
+		};
```

Comment:
Can you make this a generic utility function and move it to https://github.com/sindresorhus/file-type/blob/master/util.js ? Could be called `multiByteIndexOf` or something better.

Reply:
👍